openmc/mgxs/mgxs.py

```diff
@@ -1723,8 +1723,8 @@ def compute_xs(self):
         if self.correction == 'P0':
             scatter_p1 = self.tallies['scatter-P1']
             scatter_p1 = scatter_p1.get_slice(scores=['scatter-P1'])
-            energy_filter = openmc.Filter(type='energy')
-            energy_filter.bins = self.energy_groups.group_edges
+            energy_filter = copy.deepcopy(self.tallies['scatter'].
```

Comment:
I think it would be best to break this up into two lines:

```python
energy_filter = self.tallies['scatter']
energy_filter = copy.deepcopy(energy_filter)
```

Sometimes doing less per line is actually more IMHO. In this case, it can be easier to identify bugs in failing code when multiple methods are not chained together as they are here.